src/w3c/style.js

```diff
@@ -114,28 +114,30 @@ export function run(conf) {
   // Make sure the W3C stylesheet is the last stylesheet, as required by W3C Pub Rules.
   sub("beforesave", styleMover(finalStyleURL));
 
-  if (conf.darkMode) {
-    const darkModeStyleUrl = getStyleUrl("dark.css");
-    document.head.appendChild(
-      html`<style>
-        :root {
-          color-scheme: light dark;
-        }
-      </style>`
-    );
-    document.head.appendChild(
-      html`<meta name="color-scheme" content="light dark" />`
-    );
-    document.head.appendChild(
-      html`<link
-        rel="stylesheet"
-        href="${darkModeStyleUrl.href}"
-        media="(prefers-color-scheme: dark)"
-      />`
-    );
-    // As required by W3C Pub Rules.
-    sub("beforesave", styleMover(darkModeStyleUrl));
+  /** @type HTMLMetaElement */
+  let colorScheme = document.querySelector("head meta[name=color-scheme]");
+  if (!colorScheme) {
+    colorScheme = html`<meta name="color-scheme" content="light dark" />`;
+    document.head.appendChild(colorScheme);
   }
+  const darkModeStyleUrl = getStyleUrl("dark.css");
+  const css = `:root {
```

Comment:
If the meta tag sets this, any idea why this is needed, @sidvishnoi ? Just for completeness?

Reply:
I think we can remove this now. It existed before I added that meta tag. Though, check if any browser support differences.